rust/core-lib/src/view.rs

```diff
@@ -125,8 +126,48 @@ impl View {
         self.cursor_col = col;
     }
 
-    pub fn get_cursor_col(&mut self) -> usize {
-        self.cursor_col
+    /// Toggles a caret at the given offset.
+    pub fn toggle_sel(&mut self, offset: usize) {
+        self.sel_dirty = true;
+        if !self.selection.regions_in_range(offset, offset).is_empty() {
+            self.selection.delete_range(offset, offset);
+            if !self.selection.is_empty() {
+                return;
+            }
+        }
+        let region = SelRegion {
+            start: offset,
+            end: offset,
+            // TODO: might want to set horiz to some meaningful value.
+            horiz: None,
+            affinity: Affinity::default(),
+        };
+        self.selection.add_region(region);
+    }
+
+    /// Move the selection by the given movement. Return value is the offset of
+    /// a point that should be scrolled into view.
+    ///
+    /// If `modify` is `true`, the selections are modified, otherwise the results
+    /// of individual region movements become carets.
+    pub fn do_move(&mut self, text: &Rope, movement: Movement, modify: bool)
+        -> Option<usize>
+    {
+        let new_selection = selection_movement(movement, &self.selection, self, text,
+            modify);
+        self.selection = new_selection;
+        // We somewhat arbitrarily choose the last region for setting the old-style
+        // selection state, and for scrolling it into view if needed. This choice can
```

Comment:
I tested and what Sublime does, in order:
- If any cursor is in view, it doesn't scroll. 
- If there is a cursor above the viewport, it scrolls up until a cursor is in view.
- If there is a cursor below the viewport, it scrolls down until a cursor is in view.

I'm not sure I like the arbitrary preference for cursors above to cursors below, even if the one above is further. I think the ideal behaviour might be to scroll to the closest cursor.